buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -464,6 +464,7 @@ public static int writeUtf8(ByteBuf buf, CharSequence seq) {
 
     // Fast-Path implementation
     static int writeUtf8(AbstractByteBuf buffer, int writerIndex, CharSequence seq, int len) {
+        buffer.ensureAccessible();
```

Comment:
> @Scottmitch I was wondering if we should just check the ensureAccessible() in ensureWritable(...) wdyt ?

Just saw this message ... yes this sounds better to me. However we should revisit usage of `ensureWritable` because we may be able to change `checkIndex` to `checkIndex0`.